engine/client/gl_local.h

```diff
@@ -349,6 +349,72 @@ void R_LightForPoint( const vec3_t point, color24 *ambientLight, qboolean invLig
 int R_CountSurfaceDlights( msurface_t *surf );
 int R_CountDlights( void );
 
+
+/// <STROBE> // Move to strobe.h ?
+//#define SolidConsoleX 0
+//#define SolidConsoleY (y - scr_width->value * 3 / 4)
+//#define SolidConsoleW (scr_width->value)
+//#define SolidConsoleH (scr_width->value * 3 / 4)
+//extern wrect_t con_rect; //Float - Int incompatibility
+typedef struct _rectf_s
+{
+	float x, y, w, h;
+} _rectf_t;
+extern _rectf_t con_rect;
+
+void R_Strobe( void );
+
+typedef enum {
+	p_positive = 1, // Phase: Positive
+	p_inverted = 1 << 1, // Phase: Inverted
+	f_normal = 1 << 2 // Frame: Normal
+} fstate_e; // Frame State
+
+typedef struct StrobeInfo_s {
+	unsigned int fCounter; // Frame counter
+	unsigned int pCounter, pNCounter, pBCounter; // Positive phase counters
+	unsigned int nCounter, nNCounter, nBCounter; // Negative phase counters
+	double deviation; // Standard deviation
+	double cdTimer; // Cooldown timer
+	fstate_e frameInfo; // Frame info
+} StrobeInfo_t;
+
+extern StrobeInfo_t StrobeInfo;
+
+#define STROBE_DEVIATION_LIMIT 2.75
+#define STROBE_DEVIATION_SIZE 60
+
+// Experimental 'Badness' function
+#define STROBE_BADNESS( diffP, diffNs ) \
```

Comment:
Macros are one of the greatest thing from C language, but they are also may cause not expected bugs. 

In this case, you may pass something like: _calculatePercentage( 2 + 2, 1 ) and this will be expanded to: ( 100 * 2 + 2 / 1 ), leaving you with incorrect result.

I recommend to move them to _inline functions or just put macro arguments to brackets. For example:
#define _calculatePercentage( x, y ) ( 100 * (  y ) / ( x ) )

Another thing is that it will be hard to read such code and debug, so inlined functions is preferred. And you will get a free type checking! :)

Reply:
I wanted to make them as inline functions at first place, but because of there is no separate c file for strobe I used macros as it would be bad to put these internal functions into gl_rmain.

If you approve separating the "strobe", which in my opinion is the way to go, I will do that but I am not into project files like cmake files so I may not be able to integrate separate files into the project files properly.

Reply:
CMake and Makefiles both do recursive searching of any *.c files.

Except is Android.mk, but it's not too hard. Just add path to file at the end of LOCAL_SRC_FILES and don't forget reverse slash at the end of previous line.